commands/deploy.go

```diff
@@ -343,9 +343,42 @@ func (p *deploycmd) deployFuncV20180708(c *cli.Context, app *models.App, funcfil
 	if funcfile.Name == "" {
 		funcfile.Name = filepath.Base(filepath.Dir(funcfilePath)) // todo: should probably make a copy of ff before changing it
 	}
-	fmt.Printf("Deploying %s to app: %s\n", funcfile.Name, app.Name)
 
-	var err error
+	oracleProvider, _ := getOracleProvider()
+	if oracleProvider != nil && oracleProvider.ImageCompartmentID != "" {
+		// If the provider is Oracle and ImageCompartmentID is present, we need to deploy image to the ImageCompartmentID.
+
+		// Check Registry and build the repository name.
+		if client.Registry() == "" {
+			return errors.New("registry name must be set, be sure to configure your context file")
+		}
+		res, err := getRepositoryNamePrefix(client.Registry())
+		if err != nil {
+			return err
+		}
+		repositoryName := res + "/" + funcfile.Name
+
+		// Create artifactsClient and set region.
+		artifactsClient, err := artifacts.NewArtifactsClientWithConfigurationProvider(oracleProvider.ConfigurationProvider)
+		if err != nil {
+			return err
+		}
+		artifactsClient.SetRegion(getRegion(oracleProvider))
+
+		// Check if the repository exists in ImageCompartmentID
+		repositoryExists, err := doesRepositoryExistInCompartment(repositoryName, oracleProvider.ImageCompartmentID, artifactsClient)
+		if err != nil {
+			return err
+		}
+		if !repositoryExists {
+			createContainerRepositoryInCompartment(repositoryName, oracleProvider.ImageCompartmentID, artifactsClient)
+			if err != nil {
```

Comment:
This looks wrong. `err` hasn't been assigned in the inner block.

Reply:
My bad. This code has moved around so much that I am missing things.